src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -170,6 +168,44 @@ class Holder {
                 }
             }
             """);
+        } else {
+            String invokerName = javaName + "$invoker";
+            String invokerFactoryName = javaName + "$makeInvoker";
+            String paramExprs = paramExprs(declType, finalParamNames, isVarArg);
+            appendLines(STR."""
+                public interface \{invokerName} {
+                    \{retType} \{javaName}(\{paramExprs});
+                }
+
+                """);
+            emitDocComment(decl);
+            appendLines(STR."""
+                public static \{invokerName} \{invokerFactoryName}(MemoryLayout... layouts) {
+                    class Holder {
+                        static final FunctionDescriptor BASE_DESC = \{descriptorString(2, descriptor)};
```

Comment:
Why is the descriptor inside the holder class but the MH is not?

Reply:
Ugh - because the method handle depends on the layout parameter. But if that's the case, what's the benefit of the holder class? I'd thought that to get maximum performance, the specialized method handle had to be a constant.

Reply:
We re-link the MH every time, so it is not a constant. Note that it depends on `layouts`.

Reply:
Yes, but if a client wants to stash an invoker with given shape in a static final field, then being able to trust that MH would be important I think.

Reply:
Field of lambda classes are trusted right? So the captured MH will be a trusted instance field?

Reply:
FWIW, I had a record in an earlier iteration, but realized it wasn't needed, and using a lambda expression makes the generated code a bit more concise.

Reply:
Ok, I did a quick experiment, and the JIT doesn't seem to be able to constant fold through a captured MH field. But it does work for a record. I will switch to using a record then